### PR TITLE
OPENEUROPA-1415: Refactor FileValueObject and ValueObject classes

### PR DESCRIPTION
**Pull request by drupol**

### CODE REVIEW

File: src/ValueObject/FileValueObject.php [hunk 1]
```diff
@@ -112,127 +78,47 @@ public static function fromArray(array $values = []): ValueObjectInterface {
...
+    $clone = clone $this;
 
-    return $this;
+    $clone->storage['title'] = $title;
+
+    return $clone;
   }
 
   /**
    * Setter.
```
Comment (sergepavle) [inline src/ValueObject/FileValueObject.php]: Comment not consistent with other analogical methods.

File: tests/Unit/ValueObject/DateValueObjectTest.php [hunk 2]
```diff
@@ -21,11 +21,11 @@ public function testFromArray(array $data, array $expected) {
...
-    $this->assertEquals($expected['week_day'], $date->getWeekDay());
-    $this->assertEquals($expected['month'], $date->getMonth());
-    $this->assertEquals($expected['month_name'], $date->getMonthName());
-    $this->assertEquals($expected['year'], $date->getYear());
+    $this->assertEquals($expected['day'], $date->day());
+    $this->assertEquals($expected['week_day'], $date->week_day());
+    $this->assertEquals($expected['month'], $date->month());
+    $this->assertEquals($expected['month_name'], $date->month_name());
+    $this->assertEquals($expected['year'], $date->year());
   }
```
Comment (sergepavle) [inline tests/Unit/ValueObject/DateValueObjectTest.php]: Probably better to use the same code as in \Drupal\Tests\oe_theme\Kernel\ValueObject\DateTimeTestBase::testFromDateTimeObject
Or use $this->assertArrayEquals()

File: src/ValueObject/FileValueObject.php (hunk 1 above)
Comment (drupol) [reply src/ValueObject/FileValueObject.php]: Fixed.

File: tests/Unit/ValueObject/DateValueObjectTest.php (hunk 2 above)
Comment (drupol) [reply tests/Unit/ValueObject/DateValueObjectTest.php]: Fixed.

**Comment by ademarco** [comment]
Closing as won't fix.

**Pull request closed without merging**